### PR TITLE
Fix for issue 8578 involving non-breaking strings.

### PR DESCRIPTION
**Pull request by sg00dwin**
Creation of two mixins for breaking strings within certain markup contexts.

``` 
// Looks to break the string at acceptable break points in certain cases, 
// otherwise it will be broken at an arbitrary point.
// include min-width: 0 which is needed for firefox
.word-break() 

// Guarantees the string will break in all cases, by using word-break: break-all; 
// which is used by firefox and IE
.word-break-all()
```

Fixes https://github.com/openshift/origin/issues/8578

@spadgett @rhamilto @benjaminapetersen

### CODE REVIEW

File: assets/app/styles/_core.less [hunk 1]
```diff
@@ -693,6 +693,9 @@ select:invalid {
   .text-overflow();
 }
 
+.word-break-all {
+  .word-break-all()
+}
```
Comment (spadgett) [inline assets/app/styles/_core.less]: If you leave off the `()` when you define the style in _mixins.less, you can use it as a mixin or a normal class. Then you don't need the duplicate definition. Same comment for existing `.word-break` and `.word-break()`.

http://lesscss.org/features/#mixins-feature

**Comment by spadgett** [comment]
So this means the event messages will break mid-word even for short strings? I'm not sure how I feel about that :/

I'd almost rather use `overflow: hidden`.

**Comment by sg00dwin** [comment]
@spadgett I'll update the mixins

I'm not understanding the context of "break mid-word even for short strings". If we want to wrap a non-breaking string within a table cell it can only be done in FF or IE by using word-break:break-all

word-break-all

**Comment by spadgett** [comment]
My concern is that it will also break tiny words like "pod" if they happen to hit the edge instead of wrapping at word boundaries.

**Comment by sg00dwin** [comment]
Updated mixin and removed duplicate rules in _core.less

Small words wouldn't be wrapped since the smallest width for the message <td> is ~500px, only strings longer than that would wrap.

**Comment by liggitt** [comment]
if it falls on the edge, it would, right? something like
```
this is a really long message that breaks at about 500 pixels right he
re.
```

**Comment by sg00dwin** [comment]
ah yes it would, but only for FF and IE. 
Webkit would use the next property in the mixin word-break: break-word;

**Comment by sg00dwin** [comment]
If we used table-layout: fixed and sized the columns reasonably, we wouldn't have to use word-break:break-all to target FF, IE ... Let me try that.

**Comment by spadgett** [comment]
It would also fix the columns resizing when you filter the table.

**Comment by sg00dwin** [comment]
@spadgett updated pr
- events-table uses ```table-layout:fixed```
- name and message gets .word-break mixin (sans word-break:break-all)

File: assets/app/views/builds.html [hunk 2]
```diff
@@ -95,7 +95,7 @@
                       <span>- {{build.metadata.creationTimestamp | date : 'short'}}</span>
                     </td>
                     <td data-title="Type">{{build.spec.strategy.type}}</td>
-                    <td data-title="Source">
+                    <td data-title="Source" class="word-break-all">
```
Comment (spadgett) [inline assets/app/views/builds.html]: Do we need break-all here?

**Comment by spadgett** [comment]
bindata looks wrong

File: assets/app/views/builds.html (hunk 2 above)
Comment (sg00dwin) [inline assets/app/views/builds.html]: Yes, if the source url is long it will cause FF and IE to extend the table beyond it's container.

**Comment by sg00dwin** [comment]
Updated bindata pushed

**Comment by spadgett** [comment]
[merge]

**Pull request merged**